src/containers/SettingsPage/index.js

```diff
@@ -1,5 +1,43 @@
 import { connect } from 'react-redux'
+import { reduxForm, formValueSelector, submit } from 'redux-form'
 
 import Settings from 'components/Settings'
+import { updateSettings } from 'actions/settings'
 
-export default connect()(Settings)
+const FORM = {
+  form: 'settingsForm',
+  onChange: async (values, dispatch, props) => {
+    const { formValues, syncErrors, pristine, dirty } = props
+    // TODO should only if no errors but seems to be an issue with
+    // onchange validation
+    // https://github.com/erikras/redux-form/issues/3012
+    return await dispatch(updateSettings(formValues.settings))
```

Comment:
Oh wait, "updateSettings" already submits the form. Check the [onSubmit](http://redux-form.com/7.0.3/docs/api/ReduxForm.md/#-code-onsubmit-function-code-optional-) function if you can access the validator errors there!